### PR TITLE
Ogury Bid Adapter: add new adapter

### PR DESCRIPTION
**Pull request by jogury**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Initialization of the new Ogury Bid Adapter

- test parameters for validating bids
```
{
  bidder: 'ogury',
  params: {
      assetKey: 'OGY-CA41D116484F',
      adUnitId: '2c4d61d0-90aa-0139-0cda-0242ac120004'
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer: web.inventory@ogury.co
- [x ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
https://github.com/prebid/prebid.github.io/pull/2943

### CODE REVIEW

File: modules/oguryBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+function getFloor(bid) {
+  if (isFn(bid.getFloor)) {
+    let floorResult = bid.getFloor({
+      currency: 'USD',
+      mediaType: 'banner',
+      size: '*'
+    });
+
+    return floorResult.floor;
```
Comment (robertrmartinez) [inline modules/oguryBidAdapter.js:133]: it is possible that if floors is set in non USD, and publisher does not have currency module included / enabled, that this call will return non USD floor.

It is a good idea to verify that the currency returned is in USD in this case, just to be safe!

Maybe something like this?
```js
return floorResult.currency === 'USD' ? floorResult.floor : 0;
```

File: modules/oguryBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+function onTimeout(timeoutData) {
+  logWarn(timeoutData);
+}
+
+function onBidWon(bid) {
+}
+
+function onSetTargeting(bid) {
+  logWarn(bid);
```
Comment (robertrmartinez) [inline modules/oguryBidAdapter.js:122]: Not sure this is something we need?

Log a warning every single time there is a timeout or setTargeting?

File: modules/oguryBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        ttl: 60,
+      };
+
+      bidResponse.ad = bid.adm;
+
+      bidResponses.push(bidResponse);
+    });
+  });
+
+  if (bidResponses.length === 0) logWarn('bid responses is empty');
```
Comment (robertrmartinez) [inline modules/oguryBidAdapter.js:109]: Same here, we do not need to log a warning just because there is a no bid.

**Review by robertrmartinez — CHANGES_REQUESTED** [review]
Couple small inputs.

Polluting the console with warnings that are not actual warnings is not ideal.

File: modules/oguryBidAdapter.js (hunk 1 above)
Comment (mbouallagui) [reply modules/oguryBidAdapter.js]: I agree. I added the check you suggested and a new unit test to cover the case of unsupported currency 
it("should handle bidFloor when currency is not USD")

File: modules/oguryBidAdapter.js (hunk 2 above)
Comment (mbouallagui) [reply modules/oguryBidAdapter.js]: In fact, that was added during some debugging and is not needed anymore. I removed the logs as well as useless empty functions.

File: modules/oguryBidAdapter.js (hunk 3 above)
Comment (mbouallagui) [reply modules/oguryBidAdapter.js]: same as above.

**Comment by robertrmartinez** [comment]
@jogury Thanks for addressing the comments, please just undo the changes you made to package-lock.json!

**Comment by mbouallagui** [comment]
@robertrmartinez , I've just removed the changes in package-lock.json from the PR. Sorry about that.

**Review by patmmccann — CHANGES_REQUESTED** [review]
You must return advertiserDomains

**Comment by antoinezaz** [comment]
@patmmccann thanks for your comment, now we return the advertiserDomains

**Comment by patmmccann** [comment]
Lgtm ; thanks! @robertrmartinez is this ready to merge or any other suggestions?

**Review by robertrmartinez — APPROVED** [review]
Thanks for all the work to get this done! 👍

**Pull request merged**